### PR TITLE
Be principled about formatting commands.

### LINKED ISSUE #16: rust-mpd doesn't properly quote strings.
If you have files or tags with `"` or `\` in them, rust-mpd doesn't properly escape them in quotes.

### PR DESCRIPTION
**Pull request by tomprince**
Rather than littering the code with calls so `format!`, pass all the command arguments as structured data, so there is one place where everything can be quoted.

Fixes #16.

### CODE REVIEW

**Comment by tomprince** [comment]
I'll rebase this once if/when #12/#14 land.

**Comment by tomprince** [comment]
This is ready now.

**Comment by kstep** [comment]
Please rebase to resolve conflicts, and I will merge it. Thank you for your efforts!

File: src/status.rs [hunk 1]
```diff
@@ -70,20 +70,17 @@ impl Encodable for Status {
...
+                            e.emit_option_some(|e| {
+                                e.emit_tuple(2, |e| {
+                                    e.emit_tuple_arg(0, |e| p.0.num_seconds().encode(e))?;
+                                    e.emit_tuple_arg(1, |e| p.1.num_seconds().encode(e))?;
+                                    Ok(())
                                 })
-                            }
-                            None => e.emit_option_none(),
-                        };
-                        Ok(())
```
Comment (tomprince) [inline src/status.rs]: This is un unrelated change that fixes the error handling from #20. I can split it out if you'd like.

**Comment by tomprince** [comment]
Rebased.

**Pull request merged**